enumerables.rb

```diff
@@ -0,0 +1,332 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/Void
+p '-----------|||||--------'
+p '----------{(o o)}--------'
+p '--------ooO-(_)-Ooo------'
+# Initial comment
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    size.times { |index| yield(array[index]) }
+  end
+
+  def my_each_with_index
```

Comment:
When a block is given and it is a range, My each with index should return the range itself after calling the given block once for each element in self.
![Screen Shot 2020-07-21 at 7 21 55 PM](https://user-images.githubusercontent.com/52430951/88119957-7d36fa80-cb87-11ea-9ca9-70cf7c5adc8c.png)